data_steward/cdr_cleaner/cleaning_rules/race_ethnicity_record_suppression.py

```diff
@@ -0,0 +1,149 @@
+from common import JINJA_ENV, OBSERVATION
+from utils import pipeline_logging
+from constants.bq_utils import WRITE_TRUNCATE
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+
+LOGGER = logging.getLogger(__name__)
+
+# Keeps all the records that do not contain any observation_source_concept_id in the module description table
+DROP_RECORDS_QUERY = JINJA_ENV.from_string("""
+SELECT * FROM `{{project_id}}.{{dataset_id}}.observation`
+WHERE observation_id NOT IN (
+SELECT observation_id FROM `{{project_id}}.{{sandbox_id}}.{{sandbox_table}}`)
+""")
+
+# Selects all the records that will be dropped. The records that will be dropped contain observation_source_concept_id
+# in the module description table
+SANDBOX_RECORDS_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{sandbox_table}}` AS (
+SELECT * FROM `{{project_id}}.{{dataset_id}}.observation`
+WHERE observation_source_concept_id IN (1586151, 1586150, 1586152, 1586153, 1586154, 1586155, 1586156, 1586149))
+""")
+
+
+class RaceEthnicityRecordSuppression(BaseCleaningRule):
+    """
+    Any record with an observation_source_concept_id in the description table should be sandboxed and dropped from the
+    observation table.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'Any record with an observation_source_concept_id equal to any of these values (1586151, 1586150, ' \
+               '1586152, 1586153, 1586154, 1586155, 1586156, 1586149) will be sandboxed and dropped from ' \
+               'the observation table'
+        super().__init__(issue_numbers=['DC1365'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.CONTROLLED_TIER_DEID],
+                         affected_tables=OBSERVATION,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self, *args, **keyword_args):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+
+        sandbox_records_query = {
+            cdr_consts.QUERY:
+                SANDBOX_RECORDS_QUERY.render(
+                    project_id=self.project_id,
+                    sandbox_id=self.sandbox_dataset_id,
+                    sandbox_table=self.get_sandbox_tablenames(),
```

Comment:
This all looks great! Only thing, I noticed in the acceptance criteria in the ticket it says:

> It will use the shared sandbox naming convention.

So does this mean to use `get_sandbox_table_name` from `sandbox.py`?

Reply:
Hm, you're right, I think that's the one we need to use. @lrwb-aou is `get_sandbox_table_name` from `sandbox.py` the right one to use here? I'm getting `get_sandbox_tablenames` and `get_sandbox_table_name` confused 😩

Reply:
You are right.  They return different things.  One returns a single table name.  The other returns a list/iterable of table names in a sandbox.   We need to use the `get_sandbox_table_name()` from `sandbox.py` to name individual tables in the sandbox.  The other function is used to get an iterable of all available table sandbox table names associated with the cleaning rule.  The iterable returned (`get_sandbox_tablenames`) is used primarily for the report runner and integration tests.